src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -1453,6 +1519,60 @@ public int compare(List<ByteBuffer> a, List<ByteBuffer> b)
         }
     }
 
+    private static class AnnColumnComparator<T> extends ColumnComparator<List<ByteBuffer>>
+    {
+        private final SingleColumnRestriction.AnnRestriction restriction;
+        private final int index;
+        private final VectorType type;
+
+        private final float[] target;
+
+        // maybe cached in prepared statement
+        public AnnColumnComparator(SingleColumnRestriction.AnnRestriction restriction, int columnIndex, VectorType type)
+        {
+            this.restriction = restriction;
+            this.index = columnIndex;
+            this.type = type;
+            this.target = null;
+        }
+
+        // for actual re-ordering
+        private AnnColumnComparator(SingleColumnRestriction.AnnRestriction restriction, float[] target, int columnIndex, VectorType type)
+        {
+            this.restriction = restriction;
+            this.target = target;
+            this.index = columnIndex;
+            this.type = type;
+        }
+
+        @Override
+        public int compare(List<ByteBuffer> a, List<ByteBuffer> b)
+        {
+            Preconditions.checkNotNull(target);
+
+            VectorSimilarityFunction function = VectorSimilarityFunction.COSINE;
```

Comment:
Shouldn't this be using the similarity function defined in the Index?

Reply:
good catch, I will change it

Reply:
Actually, I am unsure how to get it from index into `SelectStatement`..

Reply:
Updated here https://github.com/datastax/cassandra/pull/626/commits/30b49687df92c4bed07401c32bfd9e63920ff341